### PR TITLE
WT-13307 Add diagnostic log message when removing log files.

### PR DESCRIPTION
**Pull request by sueloverso**
Here's one style of debug that can be added for any future failure like this ticket sees.

### CODE REVIEW

**Review by keitharnoldsmith — APPROVED** [review]
LGTM!

**Review by pmacko86 — APPROVED** [review]
LGTM

**Comment by sueloverso** [comment]
@pmacko86 and @keitharnoldsmith  I changed this quite a bit. I added a diag-only field so that we only print on change rather than output ever second into the log. I added the time because that is always useful information. I think it is worth a re-review.

**Comment by sueloverso** [comment]
The output looks like:
```
  { "lsn" : [2,896],
    "hdr_flags" : "",
    "rec_len" : 256,
    "mem_len" : 256,
    "type" : "message",
    "message" : "COMPUTE_MIN_LOGNUM: [1723568362:589445] min_lognum 2 ckpt file 2 sync file 2 backup_file 0 debug_log count0 old min 0"
  },```

File: src/include/log.h [hunk 1]
```diff
@@ -238,15 +239,18 @@ struct __wt_myslot {
...
+    uint32_t first_record;         /* Offset of first record in file */
+    wt_off_t log_written;          /* Amount of log written this period */
+                                   /*
+                                    * Log file information
+                                    */
+    uint32_t fileid;               /* Current log file number */
+    uint32_t prep_fileid;          /* Pre-allocated file number */
+    wt_shared uint32_t tmp_fileid; /* Temporary file number */
+#ifdef HAVE_DIAGNOSTIC
+    uint32_t min_fileid; /* Minimum file number needed */
```
Comment (pmacko86) [inline src/include/log.h:252]: I assume this gets initialized to 0 when the struct gets initialized, presumably using something like memset?

**Review by pmacko86 — APPROVED** [review]
LGTM, with one question in the comments.

File: src/include/log.h (hunk 1 above)
Comment (sueloverso) [reply src/include/log.h:252]: Yes, in `wti_logmgr_create` in `conn_log.c` the entire structure is allocated using `calloc`.

**Pull request merged**